drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) 2024, Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*Address by default is 0x55*/
+
+#define DT_DRV_COMPAT sitronix_cf1133
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(cf1133, CONFIG_INPUT_LOG_LEVEL);
+
+/* cf1133 used registers */
+#define FIRMWARE_VERSION    0x0
+#define STATUS_REG          0x1
+#define DEVICE_CONTROL_REG  0x2
+#define TIMEOUT_TO_IDLE_REG 0x3
+#define XY_RESOLUTION_HIGH  0x4
+#define X_RESOLUTION_LOW    0x5
+#define Y_RESOLUTION_LOW    0x6
+#define DEVICE_CONTROL_REG2 0x09
+#define FIRMWARE_REVISION_3 0x0C
+#define FIRMWARE_REVISION_2 0x0D
+#define FIRMWARE_REVISION_1 0x0E
+#define FIRMWARE_REVISION_0 0x0F
+#define FINGERS             0x10
+#define KEYS_REG            0x11
+#define XY0_COORD_H         0x12
+#define X0_COORD_L          0x13
+#define Y0_COORD_L          0x14
+#define I2C_PROTOCOL        0x3E
+#define MAX_NUM_TOUCHES     0x3F
+#define DATA_0_HIGH         0x40
+#define DATA_0_LOW          0x41
+#define MISC_CONTROL        0xF1
+#define SMART_WAKE_UP_REG   0xF2
+#define CHIP_ID             0xF4
+#define PAGE_REG            0xff
+
+/* Constants  */
+#define SITRONIX_MAX_SUPPORTED_POINT 10U
```

Comment:
Hey this just got me thinking... the driver and api only supports a single touch, how about hardcoding this to `1` and dropping the whole max touch detection logic? Unless you actually want to enable multitouch support in the future but then it should probably be optional, limiting this to 1 point would simplify few things and save a bunch of RAM on the stack from smaller buffers.

Reply:
So i went for only a single touch as you described ;-)